src/components/webview/repository.ts

```diff
@@ -27,31 +20,43 @@ const branchNameFromStr = (str: string): string => {
 	return branchName;
 };
 
+type PersistedState = {
```

Comment:
1) I would not add `null` to an object type
2) can we get rid of the possibility of a nullable `PersistedState`?

Reply:
getPersistedState reads from file system, we dont want to throw error when extension is loaded for the first time, so we return State | null, this makes sense. I can pass  just `remoteUrl ?? null ` instead of passing the whole state to repository service.

Reply:
Good question.

We can initialize properties with their "natural" neutral state (neutral as in the neutral value of on operation in the group theory):
- objects => null,
- strings => empty,
- numbers => 0
- arrays => empty list
- etc.